ttnn/cpp/ttnn/operations/core/work_split/work_split_tilize.hpp

```diff
@@ -176,14 +176,14 @@ struct FullRep {
 };
 
 inline std::vector<std::vector<BlockRep>> distribute_work(
-    const tt::tt_metal::LegacyShape& unpadded, const Padding& padding, uint32_t num_cores, uint32_t blocks_per_core, bool has_cliff, uint32_t nblocks_per_core_cliff) {
-    auto input_w = unpadded.rank() >= 4 ? unpadded[-4] : 1;
-    auto input_z = unpadded.rank() >= 3 ? unpadded[-3] : 1;
-    auto input_y = unpadded.rank() >= 2 ? unpadded[-2] : 1;
-
-    auto padding_w = unpadded.rank() >= 4 ? padding[padding.get_normalized_index(-4)].back : 0;
-    auto padding_z = unpadded.rank() >= 3 ? padding[padding.get_normalized_index(-3)].back : 0;
-    auto padding_y = unpadded.rank() >= 2 ? padding[padding.get_normalized_index(-2)].back : 0;
+    const ttnn::SimpleShape& logical_shape, const Padding& padding, uint32_t num_cores, uint32_t blocks_per_core, bool has_cliff, uint32_t nblocks_per_core_cliff) {
+    auto input_w = logical_shape.rank() >= 4 ? logical_shape[-4] : 1;
+    auto input_z = logical_shape.rank() >= 3 ? logical_shape[-3] : 1;
+    auto input_y = logical_shape.rank() >= 2 ? logical_shape[-2] : 1;
+
+    auto padding_w = logical_shape.rank() >= 4 ? padding[padding.get_normalized_index(-4)].back : 0;
```

Comment:
nit: should we add a check that padding has same rank as logical_shape? or check whether padding has rank larger than 4? I see, that it was done before this way, but thinking whether it's a corner case (as an example: logical shape has rank 4 and padding has rank 2 that covers only first two dimensions).

Feel free to ignore, as this is not related to your change.

Reply:
Thats reasonable. Added TT_FATAL check

Reply:
Hah, it was actually not reasonable. Removed the check..